File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        elementCount = {}
        majorityElementSize = len(nums)/2
        for num in nums:
            if num in elementCount:
                elementCount[num] += 1
            else:
                elementCount[num] = 1

            if elementCount[num] > majorityElementSize:
                    return num

File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let elementCount = {};
    let majorityElementSize = nums.length/2;
    for (const num of nums){
        if (elementCount[num] != undefined ||  elementCount[num] != null) {
            elementCount[num]++;
        } else {
            elementCount[num] = 1;
        }

        if (elementCount[num] > majorityElementSize) {
            return num;
        }         
    };
        
};